File: Braintree/src/main/java/com/braintreepayments/api/models/BraintreeRequestCodes.java
package com.braintreepayments.api.models;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Request codes that Braintree uses when communicating between {@link android.app.Activity}s, and
 * {@link android.app.Fragment}s.
 */
@Retention(RetentionPolicy.SOURCE)
@IntDef({
        BraintreeRequestCodes.THREE_D_SECURE,
        BraintreeRequestCodes.VENMO,
        BraintreeRequestCodes.ANDROID_PAY,
        BraintreeRequestCodes.PAYPAL,
        BraintreeRequestCodes.VISA_CHECKOUT,
        BraintreeRequestCodes.GOOGLE_PAYMENT,
        BraintreeRequestCodes.IDEAL,
        BraintreeRequestCodes.SAMSUNG_PAY
})
public @interface BraintreeRequestCodes {
    int THREE_D_SECURE = 13487;
    int VENMO = 13488;
    int ANDROID_PAY = 13489;
    int PAYPAL = 13591;
    int VISA_CHECKOUT = 13592;
    int GOOGLE_PAYMENT = 13593;
    int IDEAL = 13594;
    int SAMSUNG_PAY = 13595;
}

File: BraintreeCore/src/main/java/com/braintreepayments/api/core/BraintreeRequestCodes.kt
package com.braintreepayments.api.core

import androidx.annotation.RestrictTo

private const val THREE_D_SECURE_CODE = 13487
private const val VENMO_CODE = 13488
private const val PAYPAL_CODE = 13591
private const val VISA_CHECKOUT_CODE = 13592
private const val GOOGLE_PAY_CODE = 13593
private const val LOCAL_PAYMENT_CODE = 13596
private const val SEPA_DEBIT_CODE = 13597

/**
 * Request codes that Braintree uses when communicating between Android Activities and Fragments.
 */
@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
enum class BraintreeRequestCodes(val code: Int) {
    THREE_D_SECURE(THREE_D_SECURE_CODE),
    VENMO(VENMO_CODE),
    PAYPAL(PAYPAL_CODE),
    VISA_CHECKOUT(VISA_CHECKOUT_CODE),
    GOOGLE_PAY(GOOGLE_PAY_CODE),
    LOCAL_PAYMENT(LOCAL_PAYMENT_CODE),
    SEPA_DEBIT(SEPA_DEBIT_CODE),
}
